src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const propAfterAction = [];
+
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties' :
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case action.type === 'removeProperties' :
+        for (const keys of action.keysToRemove) {
+          if (stateCopy.hasOwnProperty(keys)) {
+            delete stateCopy[keys];
+          };
```

Comment:
I think there is no need to check 'hasOwnProperty', it won't be deleted if it isn't there:)